#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, bool_t, double_t, int_t

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "Whether to apply this plugin or not", True)
gen.add("only_proxemics", bool_t, 0, "calculate all the features or only the proxemics feature.", False)
gen.add("w1", double_t, 0, "weight for the goal distance feature", 0.1, 0.0, 1.0);
gen.add("w2", double_t, 0, "weight for the obstacle distance feature", 0.5, 0.0, 1.0);
gen.add("w3", double_t, 0, "weight for the proxemics feature", 0.4, 0.0, 1.0);
gen.add("w4", double_t, 0, "weight for the proxemics back feature", 0.0, 0.0, 1.0);
gen.add("w5", double_t, 0, "weight for the proxemics right side feature", 0.0, 0.0, 1.0);

exit(gen.generate("upo_social_layer", "upo_social_layer", "SocialPlugin"))
